android/java/org/chromium/chrome/browser/playlist/settings/BravePlaylistPreferences.java

```diff
@@ -0,0 +1,201 @@
+        mEnablePlaylistSwitch = (ChromeSwitchPreference) findPreference(PREF_ENABLE_PLAYLIST);
+        mEnablePlaylistSwitch.setChecked(
+                SharedPreferencesManager.getInstance().readBoolean(PREF_ENABLE_PLAYLIST, true));
+        mEnablePlaylistSwitch.setOnPreferenceChangeListener(this);
+
+        mAddToPlaylistButtonSwitch =
+                (ChromeSwitchPreference) findPreference(PREF_ADD_TO_PLAYLIST_BUTTON);
+        mAddToPlaylistButtonSwitch.setChecked(SharedPreferencesManager.getInstance().readBoolean(
+                PREF_ADD_TO_PLAYLIST_BUTTON, true));
+        mAddToPlaylistButtonSwitch.setOnPreferenceChangeListener(this);
+
+        mAutoPlaySwitch = (ChromeSwitchPreference) findPreference(PREF_AUTO_PLAY);
+        mAutoPlaySwitch.setChecked(
+                SharedPreferencesManager.getInstance().readBoolean(PREF_AUTO_PLAY, true));
+        mAutoPlaySwitch.setOnPreferenceChangeListener(this);
+
+        mAutoSaveMediaForOfflinePreference =
+                (Preference) findPreference(PREF_AUTO_SAVE_MEDIA_FOR_OFFLINE);
+        updateAutoSaveMedia();
+
+        mStartPlaybackSwitch = (ChromeSwitchPreference) findPreference(PREF_START_PLAYBACK);
+        mStartPlaybackSwitch.setChecked(
+                SharedPreferencesManager.getInstance().readBoolean(PREF_START_PLAYBACK, true));
+        mStartPlaybackSwitch.setOnPreferenceChangeListener(this);
+
+        mResetPlaylist = (BravePlaylistResetPreference) findPreference(PREF_RESET_PLAYLIST);
+        mResetPlaylist.setOnPreferenceClickListener(preference -> {
+            if (mPlaylistService != null) {
+                mPlaylistService.resetAll();
+                PlaylistPreferenceUtils.resetPlaylistPrefs(getActivity());
+                BraveRelaunchUtils.askForRelaunch(getActivity());
+            }
+            return true;
+        });
+
+        updatePlaylistSettingsState(
+                SharedPreferencesManager.getInstance().readBoolean(PREF_ENABLE_PLAYLIST, true));
+    }
+
+    @Override
+    public void onDestroy() {
+        if (mPlaylistService != null) {
+            mPlaylistService.close();
+        }
+        super.onDestroy();
+    }
+
+    @Override
+    public void onConnectionError(MojoException e) {
+        mPlaylistService = null;
+        initPlaylistService();
+    }
+
+    private void initPlaylistService() {
+        if (mPlaylistService != null) {
+            return;
+        }
+
+        mPlaylistService = PlaylistServiceFactoryAndroid.getInstance().getPlaylistService(this);
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        getActivity().setTitle(R.string.brave_playlist);
+        SettingsUtils.addPreferencesFromResource(this, R.xml.brave_playlist_preferences);
+        initPlaylistService();
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        new Handler().post(() -> { updateAutoSaveMedia(); });
```

Comment:
Why it is required to set `mAutoSaveMediaForOfflinePreference` after activity resume?
Can the `PREF_AUTO_SAVE_MEDIA_FOR_OFFLINE` pref be changed outside of `BravePlaylistPreferences` class?

Reply:
@AlexeyBarabash i have added the update on resume to be sure of the changes. and once we make the change from the selection page and when we get back to the main screen, it should also update the summary with selected option.